Boa.Constrictor/Screenplay/Conditions/Collection/IsACollectionOfType.cs

```diff
@@ -0,0 +1,95 @@
+namespace Boa.Constrictor.Screenplay
+{
+    /// <summary>
+    /// Methods to check if the collection items satifies the given conditions.
+    ///</summary>
+    ///<example>
+    /// Sample usage:
+    /// Actor.WaitsUntil(TextList.For(ResultPage.ResultLinks), IsAcollectionOfType{string}.AndEveryItem(IsEqualTo.Value(link)));
+    ///</example>
+    public class IsACollectionOfType<T>
+    {
+        #region Methods 
+
+        /// <summary>
+        /// Checks for a condition in the size of a collection
+        /// </summary>
+        /// <example>
+        /// Actor.WaitsUntil(TextList.For(ResultPage.ResultLinks), IsAcollectionOfType{string}.AndHasSizeThat(IsEqualTo.Value(2)));
+        /// </example>
+        /// <param name="condition">The condition to evaluate.</param>
+        /// <returns>HasSize</returns>
+        public static HasSize<T> AndHasSizeThat(ICondition<int> condition)
+        {
+            return new HasSize<T>(condition);
+        }
+
+        /// <summary>
+        /// Checks for a condition in every item of a collection
+        /// </summary>
+        /// <example>
+        /// Actor.WaitsUntil(TextList.For(ResultPage.ResultLinks), IsAcollectionOfType{string}.AndEveryItem(IsEqualTo.Value(link)));
+        /// </example>
+        /// <param name="condition">The condition to evaluate.</param>
+        /// <returns>EveryItem</returns>
+        public static EveryItem<T> AndEveryItem(ICondition<T> condition)

```

Comment:
Rather than using "And" in these builder methods, I think "Where" may be a better word. "Where" harkens to Linq, and it implies a continuation of the Condition instead of a conjunction with a new assertion.